TracerLib/TracerImpl/Tracer.cs

```diff
@@ -68,8 +68,7 @@ public void StopTrace()
 
```

Comment:
StopTrace() method -> line with "_methodsStack.Pop().Data.Time.Stop(); ;" contains extra ';' symbol.

Reply:
Method 'StartTrace()' - in previous implementation it just starts timer and collect whole information in StopTrace() method. It was right way, because we need to get clear time of working on a method without adding additional time that was spent in tracer for collecting data or building tree of calling.

Reply:
As I see you decided to leave building a tree in tracer. But rewrite moving a pointer to the 'current' branch. Am I right? 
Just imagine if you are adding one more thread which using a Tracer. Will it work properly in this case? 
What you will need to change in your implementation in this case?

Reply:
For multihreading tracer i think that i have to create entity with name like ThreadInfo which will contain fields to save tree and stack for each thread in application and in Tracer for current thread i will write information in ThreadInfo and save it in collection like Directory<>. Am i right?

Reply:
It's not so major where you will save information. 
The point is: how you are going to separate the information you catch by Tracer. 
Your case with dictionary with a key by ThreadId is acceptable.